### ISSUE TITLE
treat (quote foo/bar) the same as 'foo/bar with respect to used namespaces

### ISSUE THREAD

**Issue by pgj** [borkdude/clj-kondo#210]
**version**

`clj-kondo v2019.05.26-alpha`

**platform**

macOS

**problem**

Quoted names are not considered a reference to a namespace when the quote symbol is used.  In the meantime, it works if the `quote` function is used, which is equal to using the `'` prefix.

**repro**

```sh
$ echo "(ns foo (:require [clojure.string :as str])) 'str/str" | clj-kondo --lint -
<stdin>:1:20: warning: namespace clojure.string is required but never used
linting took 35ms, errors: 0, warnings: 1
```

however:

```sh
$ echo "(ns foo (:require [clojure.string :as str])) (quote str/str)" | clj-kondo --lint -
linting took 16ms, errors: 0, warnings: 0
```

**expected behavior**

I guess `clj-kondo` should not make a difference between `'` and `quote`, so the example above would work just like this:

```sh
$ echo "(ns foo (:require [clojure.string :as str])) 'str/str" | clj-kondo --lint -
linting took 42ms, errors: 0, warnings: 0
```

**actual behavior**

See **repro**.

**Comment by borkdude** [comment]
@pgj Using a quoted symbol doesn't actually require the namespace to be required:

```
clj
Clojure 1.10.0
user=> 'str/str
str/str
user=> (quote str/str)
str/str
```

So I think it's `(quote str/str)` that needs to be fixed, not the first example that you provided?

**Comment by pgj** [comment]
Yes, that can be possible.

**Comment by borkdude** [comment]
This example is correct:
```
echo "(ns foo (:require bar)) 'bar/bar" | clj -A:clj-kondo --lint -
<stdin>:1:19: warning: namespace bar is required but never used
```
because you do not need to require the namespace `bar` in order to produce the symbol `bar/bar`.

The same goes for these examples:

```
echo '(ns foo (:require bar)) `bar/bar' | clj -A:clj-kondo --lint -
echo '(ns foo (:require bar)) (quote bar/bar)' | clj -A:clj-kondo --lint -
```
but these do not give the warning, which I believe is the incorrect behavior.

**Comment by borkdude** [comment]
However:

Maybe in the syntax-quote case it's good to choice to count the namespace as used. E.g. in writing macros, you’ll often want to load the namespaces, even if you’re only generating code that will use those namespaces and you’re not using them at macro definition time:

```
(ns foo (:require bar))
(defmacro foo [] `(bar/do-something))
```

**Comment by borkdude** [comment]
Fixed with 9bb2e4dbbbe912a2eee3b8fdddf898b67668d773
